flink-connector-obkv-hbase/src/main/java/com/oceanbase/connector/flink/sink/OBKVHBaseStatementExecutor.java

```diff
@@ -0,0 +1,150 @@
+    private final OBKVHBaseConnectionProvider connectionProvider;
+
+    private final List<RowData> buffer = new ArrayList<>();
+
+    private SinkWriterMetricGroup metricGroup;
+
+    public OBKVHBaseStatementExecutor(
+            OBKVHBaseStatementOptions options,
+            OBKVHBaseTableSchema tableSchema,
+            OBKVHBaseConnectionProvider connectionProvider) {
+        this.options = options;
+        this.tableSchema = tableSchema;
+        this.connectionProvider = connectionProvider;
+    }
+
+    @Override
+    public void setContext(Sink.InitContext context) {
+        this.metricGroup = context.metricGroup();
+    }
+
+    @Override
+    public void addToBatch(RowData record) {
+        synchronized (buffer) {
+            buffer.add(record);
+        }
+        metricGroup.getIOMetricGroup().getNumRecordsInCounter().inc();
+    }
+
+    @Override
+    public void executeBatch() throws Exception {
+        synchronized (buffer) {
+            int count = 0;
+            List<Put> putList = new LinkedList<>();
+            List<Delete> deleteList = new LinkedList<>();
+            for (RowData record : buffer) {
+                if (record.getRowKind() == RowKind.INSERT
+                        || record.getRowKind() == RowKind.UPDATE_AFTER) {
+                    Put put = getPut(record);
+                    if (put != null) {
+                        putList.add(put);
+                    }
+                } else {
+                    Delete delete = getDelete(record);
+                    if (delete != null) {
+                        deleteList.add(delete);
+                    }
+                }
+                count++;
+                if (count >= options.getBatchSize()) {
+                    if (!putList.isEmpty()) {
+                        connectionProvider.getTable().put(putList);
+                    }
+                    if (!deleteList.isEmpty()) {
+                        connectionProvider.getTable().delete(deleteList);
+                    }
+                    metricGroup.getIOMetricGroup().getNumRecordsOutCounter().inc(count);
+                    count = 0;
+                }
+            }
+            if (count > 0) {
+                connectionProvider.getTable().put(putList);
+                connectionProvider.getTable().delete(deleteList);
+                metricGroup.getIOMetricGroup().getNumRecordsOutCounter().inc(count);
+            }
+            buffer.clear();
+        }
+    }
+
+    private Put getPut(RowData rowData) {
+        Object rowKey = tableSchema.getRowKeyFieldGetter().getFieldOrNull(rowData);
+        if (rowKey == null) {
+            return null;
+        }
+        Put put = new Put((byte[]) (rowKey));
+        for (Map.Entry<String, Map<String, RowData.FieldGetter>> entry :
+                tableSchema.getFamilyQualifierFieldGetterMap().entrySet()) {
+            String familyName = entry.getKey();
+            byte[] family = Bytes.toBytes(familyName);
```

Comment:
How connector ensure that there is only one family?
(Users should be notified that we only support one family now if it is the user who guarantees this limitation.)

Reply:
There can be multiple families in one table, here multiple families will represent multiple entries in the familyQualifierFieldGetterMap.